src/viking.js

```diff
@@ -1,14 +1,116 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor(health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+
+  attack(){
+    return this.strength;
+  }
+
+  receiveDamage(damage){
+    this.health -= damage ;
+  }
+
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+
+  constructor(name, health, strength){
+    super(health, strength);
+    this.name = name;
+  }
+
+  receiveDamage(damage){
+    this.health -= damage ;
+    if(this.health > 0) return `${this.name} has received ${damage} points of damage`;
+    else return `${this.name} has died in act of combat`
+  }
+
+  battleCry(){
+    return "Odin Owns You All!" ;
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+
+  constructor(health,strength){
+    super(health, strength);
+  }
+
+  receiveDamage(damage){
+    this.health -= damage ;
+    if(this.health !== 0) return `A Saxon has received ${damage} points of damage`;
```

Comment:
If you change this condition to this` if(this.health > 0)` , test is passing ;)

Reply:
Thank you very much for your help..this part was very confusing to me